src/main/java/gyro/google/cloudsql/DatabaseInstanceResource.java

```diff
@@ -0,0 +1,950 @@
+        }
+
+        if (getBackendType() != null) {
+            databaseInstance.setBackendType(getBackendType());
+        }
+
+        if (getDatabaseVersion() != null) {
+            databaseInstance.setDatabaseVersion(getDatabaseVersion());
+        }
+
+        if (getDiskEncryptionConfiguration() != null) {
+            databaseInstance.setDiskEncryptionConfiguration(getDiskEncryptionConfiguration().toDiskEncryptionConfiguration());
+        }
+
+        if (getDiskEncryptionStatus() != null) {
+            databaseInstance.setDiskEncryptionStatus(getDiskEncryptionStatus().toDiskEncryptionStatus());
+        }
+
+        if (getFailoverReplica() != null) {
+            databaseInstance.setFailoverReplica(getFailoverReplica().toFailoverReplica());
+        }
+
+        if (getGceZone() != null) {
+            databaseInstance.setGceZone(getGceZone());
+        }
+
+        if (getInstanceType() != null) {
+            databaseInstance.setInstanceType(getInstanceType());
+        }
+
+        if (getIpAddresses() != null && !getIpAddresses().isEmpty()) {
+            databaseInstance.setIpAddresses(getIpAddresses().stream()
+                .map(DbIpMapping::toIpMapping)
+                .collect(java.util.stream.Collectors.toList()));
+        }
+
+        if (getMasterInstance() != null) {
+            databaseInstance.setMasterInstanceName(getMasterInstance().getConnectionName());
+        }
+
+        if (getName() != null) {
+            databaseInstance.setName(getName());
+        }
+
+        if (getOnPremisesConfiguration() != null) {
+            databaseInstance.setOnPremisesConfiguration(getOnPremisesConfiguration().toOnPremisesConfiguration());
+        }
+
+        if (getProject() != null) {
+            databaseInstance.setProject(getProject());
+        }
+
+        if (getRegion() != null) {
+            databaseInstance.setRegion(getRegion());
+        }
+
+        if (getReplicaConfiguration() != null) {
+            databaseInstance.setReplicaConfiguration(getReplicaConfiguration().toReplicaConfiguration());
+        }
+
+        if (getRootPassword() != null) {
+            databaseInstance.setRootPassword(getRootPassword());
+        }
+
+        if (getScheduledMaintenance() != null) {
+            databaseInstance.setScheduledMaintenance(getScheduledMaintenance().toSqlScheduledMaintenance());
+        }
+
+        if (getSecondaryGceZone() != null) {
+            databaseInstance.setSecondaryGceZone(getSecondaryGceZone());
+        }
+
+        if (getSettings() != null) {
+            databaseInstance.setSettings(getSettings().toSettings());
+        }
+
+        if (getSwitchTransactionLogsToCloudStorageEnabled() != null) {
+            databaseInstance.setSwitchTransactionLogsToCloudStorageEnabled(getSwitchTransactionLogsToCloudStorageEnabled());
+        }
+
+        client.instances().insert(getProjectId(), databaseInstance);
```

Comment:
are there other resources that might need this resource ?
Asking to know if a thought to a wait function was given

Reply:
Added waiters